App/Commands/CommandController.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace App\Commands;
+
+use App\Console\{InputConsole, InputValidator, OutputConsole};
+use App\Interface\{BaseCommandInterface , FileManagerInterface};
+use App\Model\Fast;
+
+
+class CommandController implements BaseCommandInterface
+{
+
+    public bool $appRunning = true;
+
+    protected bool $activeFasts = false;
+
+    protected bool $canCreate = false;
+
+    protected bool $noCondition = true;
+
+    protected array $availableCommands = [];
+
+    private array $menu = [
+        [
+            'option' => 'Check the fast status',
+            'command' => CheckCommand::class,
+            'condition' => 'noCondition'
+        ],
+        [
+            'option' => 'Create Fast',
+            'command' => CreateCommand::class,
+            'condition' => 'canCreate'
+        ],
+        [
+            'option' => 'Update an active fast',
+            'command' => UpdateCommand::class,
+            'condition' => 'activeFasts',
+
+        ],
+        [
+            'option' => 'End an active fast',
+            'command' => EndCommand::class,
+            'condition' => 'activeFasts',
+
+        ],
+        [
+            'option' => 'List all fasts',
+            'command' => ListCommand::class,
+            'condition' => 'noCondition'
+        ],
+        [
+            'option' => 'Exit',
+            'command' => ExitCommand::class,
+            'condition' => 'noCondition'
+        ],
+
+    ];
+
+    public function __construct(
```

Comment:
Why do you need a constructor with empty body ?

Reply:
The body of the constructor is empty because the properties of the class are defined in the entry parameters. According to PHP 8.0 documentation you can declare class properties in the constructor with no need of assigning them in the body PHP does it for you. Am i wrong ? Here is where i read that https://stitcher.io/blog/constructor-promotion-in-php-8